Update security.md (#4001)
dart-lang/site-www: src/security.md

This page describes how to report any vulnerabilities that you find,
and lists best practices to minimize the risk of introducing a vulnerability.

> ## Security philosophy
> Flutter security strategy is based on five key pillars

Reviewer: Should we refer to this as Dart's security strategy here? Or if needed, "Dart and Flutter's"?

Perhaps formatted like this too:
```suggestion
## Security philosophy

Dart's security strategy is based on five key pillars:
```

This page describes how to report any vulnerabilities that you find,
and lists best practices to minimize the risk of introducing a vulnerability.

## Security philosophy

Dart's security strategy is based on five key pillars:

> * **Identify**: Track & prioritize key security risks by identifying core assets, 

Reviewer 2: White-space for these bullets seems a bit inconsistent

> * **Identify**: Track & prioritize key security risks by identifying core assets, 
> key threats and vulnerabilities. 
> * **Detect**: Detect and identify vulnerabilities using techniques and tools like
>  vulnerability scanning,
>   static application security testing and fuzzing.
> * **Protect**: Eliminate risks by mitigating known vulnerabilities and protect 
> critical assets against source threats.
> * **Respond**: Define processes to report, triage and respond to vulnerabilities 
> or attacks.
> * **Recover**: Build capabilities to contain and recover from an incident with 
> minimal impact.

Reviewer: Suggestions to fix the whitespace inconsistencies @mit-mit mentioned and stay in our recommended width with semantic breaks.

```suggestion
* **Identify**: Track & prioritize key security risks by identifying 
  core assets, key threats and vulnerabilities.
* **Detect**: Detect and identify vulnerabilities using techniques and tools
  like vulnerability scanning, static application security testing, and fuzzing.
* **Protect**: Eliminate risks by mitigating known vulnerabilities and
  protect critical assets against source threats.
* **Respond**: Define processes to report, triage, and
  respond to vulnerabilities or attacks.
* **Recover**: Build capabilities to contain and recover from an incident
  with minimal impact.
```

## Reporting vulnerabilities

To report a security issue, use [https://g.co/vulnz][].
Coordination and disclosure happen in the [dart-lang GitHub repos][repos]
(including [GitHub security advisories][]).

> Please include proper description of the issue,

Reviewer 2: s/proper/detailed/?

> Please include proper description of the issue,

Reviewer: ```suggestion
Please include a detailed description of the issue,
```

> Please include detailed description of the issue,

Reviewer: ```suggestion
Please include a detailed description of the issue,
```

the steps you took to create the issue, affected versions, and any 
mitigations for the issue.
The Google Security Team will respond within 5 working days of
your report on g.co/vulnz.

For more information about how Google handles security issues, see
[Google’s security philosophy][].

> ##  Flagging existing issues as security-related
> If you believe that an existing issue is security-related, 

Reviewer: ```suggestion
##  Flagging existing issues as security-related

If you believe that an existing issue is security-related, 
```

we ask that you report it via [https://g.co/vulnz][] and include 
the issue id in your report.

## Supported versions

We commit to publishing security updates for the version of Dart 

> currently on the `stable` branch.

Reviewer 2: Cross-link "stable branch" to https://dart.dev/get-dart#release-channels

Reviewer 2: Maybe s/for the version of Dart currently on/for the most recent stable Dart release/ ?

> We commit to publishing security updates for the version of Dart 
> currently on the `stable` branch.

Reviewer: ```suggestion
We commit to publishing security updates 
for the most recent [stable][] Dart release.

[stable]: /get-dart#release-channels
```

> the most recent [stable](https://dart.dev/get-dart#release-channels) Dart release branch.

Reviewer: "release branch" seems a bit redundant. 

```suggestion
the most recent [stable][] Dart release.

[stable]: https://dart.dev/get-dart#release-channels
```

## Expectations

We treat security issues equivalent to a P0 priority level and release a beta or

> hotfix for any major security issues found in the most recent stable version of 

Reviewer 2: We typically use the terminology "patch release" rather than "hotfix"

> We treat security issues equivalent to a P0 priority level and release a beta or
> hotfix for any major security issues found in the most recent stable version of 
> our SDK. 

Reviewer: - This includes @mit-mit's suggestion of "patch release" 
- Changes "stable version" -> "stable release" to match other mentions on the page
- Changes "our SDK" to "Dart SDK"

```suggestion
We treat security issues equivalent to a P0 priority level 
and release a beta or patch fix
for any major security issues found
in the most recent stable release of the Dart SDK.
```

Any vulnerability reported for Dart websites like dart.dev does not 
require a release and will be fixed in the website itself.

Dart does not have a bug bounty program.

## Receiving security updates

The best way to receive security updates is to subscribe to the 

> [dart-announce](https://groups.google.com/a/google.com/g/dart-announce) 

Reviewer 2: @parlough do we not have a site variable for dart-announce?

Reviewer: Not at the moment. Taking a look, it seems we should consider linking to it from elsewhere as well, so I can follow up with a site variable change.

Reviewer: One thing though, is the link here an internal link? I don't seem to have access.

This one works for me though:
```suggestion
[dart-announce](https://groups.google.com/a/dartlang.org/g/announce) 
```

Author: Yes its internal. Never knew about the external one. Thanks for the confirming

mailing list.

> We will also announce security updates in the technical release blog post.

Reviewer 2: While that has been the tradition, I wonder if we should consider not strictly guaranteeing it.

Author: Why not if we have already been doing it?

Reviewer 2: Because I'm not sure the few times we've done it generalize to it always being a good idea to do it. For example, I can imagine that we issue a new patch stable release to fix an issue with no blog post.

Reviewer 3: Are we intending to have some way of supporting CVE reporting in other ways?

Reviewer 2: Yup! The [GitHub advisories](https://github.com/dart-lang/sdk/security/advisories?state=published) we publish are actually all CVEs, so they are available in CVE databases, e.g. https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2022-0451

Author: Lets put this as optional
